SparseGrids/tsgLinearSolvers.cpp

```diff
@@ -164,7 +164,26 @@ std::vector<double> TasmanianTridiagonalSolver::getSymmetricEigenvalues(int n, s
     return result;
 }
 
-void TasmanianTridiagonalSolver::decompose(int n, std::vector<double> &d, std::vector<double> &e, std::vector<double> &z){
+void TasmanianTridiagonalSolver::decompose(std::vector<double> &diag, std::vector<double> &off_diag, const double mu0,
+                                           std::vector<double> &nodes, std::vector<double> &weights) {
+    switch(TasmanianTridiagonalSolver::decompose_version) {
+        case 1 :
+            weights.resize(diag.size());
```

Comment:
This is messy.
* I think you want to use `weights = std::vector<double>(diag.size(), 0.0);` immediately followed by `weights[0] = mu0`.
    * resize will not initialize the vector to zero
    * the constructor uses `0.0` as initializer by default, but we explicitly need `0.0` here so it's better to say `0.0`
    * you don't want to be jumping between `weights` and `nodes` when setting variables

Reply:
Fixed.